runtime/assembly/api/serialiser.ts

```diff
@@ -17,22 +17,22 @@ export class Serialiser {
      * Returns pointer to and size of the buffer, respectively
      * @param ptr_size runtime pointer size as specified by Polkadot Host API
      */
-    static separate_pointer_size(ptr_size: u64): i32[] {
-        let value_ptr: i32 = i32(ptr_size);
-        let value_size: i32 = i32((ptr_size >> 32));
-        return [value_ptr, value_size];
+    static separatePointerSize(ptr_size: u64): i32[] {
+        let valuePtr: i32 = i32(ptr_size);
+        let valueSize: i32 = i32((ptr_size >> 32));
+        return [valuePtr, valueSize];
     }
 
     /**
      * Serialises the arguments into u64 number containing the pointer and the length of the bytes
-     * @param value_ptr 
+     * @param valuePtr 
      * @param value_len 
      */
-    static serialise_result(result: u8[]): u64 {
-        let value_ptr = result.dataStart;
-        let value_size = result.length;
+    static serialiseResult(result: u8[]): u64 {
```

Comment:
When we are using `serialiseResult` in the Storage module, do we need to `__retain` the `valuePtr`?
What if we remove the `__retain` will it work correctly?
We need to use `__retain` when we are returning values to the Rust env, but for calling into imported WASM function, do we need to do that? Using that Retain we might be leaking memory. My guess is that `wasmi` handles clearing out the data that is passed but when we are calling into imported function I doubt that someone is clearing the memory. 

Just something that we can try. If it works without `__retain` we might want to have 2 versions of the serialise -> one that retains and one that does not.